maven-api-impl/src/main/java/org/apache/maven/internal/impl/model/DefaultModelBuilder.java

```diff
@@ -564,20 +564,21 @@ public void mergeRepositories(List<Repository> toAdd, boolean replace) {
         // Infer inner reactor dependencies version
         //
         Model transformFileToRaw(Model model) {
-            List<Dependency> newDeps = new ArrayList<>();
-            boolean modified = false;
+            if (model.getDependencies().isEmpty()) {
+                return model;
+            }
+            List<Dependency> newDeps = new ArrayList<>(model.getDependencies().size());
             for (Dependency dep : model.getDependencies()) {
                 if (dep.getVersion() == null) {
-                    Dependency.Builder depBuilder = null;
                     Model depModel = getRawModel(model.getPomFile(), dep.getGroupId(), dep.getArtifactId());
```

Comment:
@gnodet I think this part of the code (starting here or below `if (depModel != null)`) is a good candidate for extraction to improve readability. It seems to just apply the new logic to retrieve the version from the aggregated build. However, I do not have a clear name for that method since I'm not familiar with the common terminology. Do you have a suggestion?

Reply:
Good point, I've extracted it in https://github.com/apache/maven/pull/1822